controller/codebase.go

```diff
@@ -107,11 +107,18 @@ func (c *CodebaseController) Edit(ctx *app.EditCodebaseContext) error {
 	var existingWorkspaces []*app.Workspace
 	for _, workspace := range workspaces {
 		openLink := rest.AbsoluteURL(ctx.Request, fmt.Sprintf(app.CodebaseHref(cb.ID)+"/open/%v", workspace.Config.Name))
+
+		ideLink := workspace.GetHrefByRelOfWorkspaceLink("ide url")
+		selfLink := workspace.GetHrefByRelOfWorkspaceLink("self link")
```

Comment:
should there be a const for `self link` ?

Reply:
thanks. will fix